packages/core/src/connectivity/scp-cf/environment-accessor.ts

```diff
@@ -245,86 +245,95 @@ export function extractClientCredentials(
 }
 
 function selectXsuaaInstance(token?: JwtPayload): XsuaaServiceCredentials {
-  const xsuaaInstances = getServiceList('xsuaa');
+  const xsuaaInstances: XsuaaService[] = getServiceList('xsuaa') as any;
 
-  if (!xsuaaInstances.length) {
+  return (
+    selectViaJwt(xsuaaInstances, token) || selectWithoutJwt(xsuaaInstances)
+  );
```

Comment:
I would expect, that I either have a JWT or not and act based on this.
```suggestion
  return token ? selectViaJwt(xsuaaInstances, token) : selectWithoutJwt(xsuaaInstances);
```

Currently, if I have a token and no xsuaa instance could be matched, we go over to selection without jwt, where the error message states that there is no jwt ("and no JWT is given to decide which one to use").

Also, if I have no jwt, in the current flow I have to call the "with jwt" function although it isn't really useful, which is a bit confusing.

Consider adding back a function like `takeFirstAndWarn` that handles the case generically that there are multiple xsuaa instances matched (independently of whether there was a jwt or not).